src/lib_json/json_reader.cpp

```diff
@@ -1639,6 +1671,41 @@ bool OurReader::decodeNumber(Token& token, Value& decoded) {
   return true;
 }
 
+bool OurReader::decodeHexadecimal(Token& token) {
+  Value decoded;
+  if (!decodeHexadecimal(token, decoded))
+    return false;
+  currentValue().swapPayload(decoded);
+  currentValue().setOffsetStart(token.start_ - begin_);
+  currentValue().setOffsetLimit(token.end_ - begin_);
+  return true;
+}
+
+bool OurReader::decodeHexadecimal(Token& token, Value& decoded) {
+  Location current = token.start_;
+  if (current < token.end_ && *current == '0') ++current;
+  if (current < token.end_ && *current == 'x') ++current;
+  Json::LargestUInt value = 0;
+  if (current >= token.end_)
+    return addError("Zero hexadecimal digits.", token);
+  if (current + (sizeof(value) * 2) < token.end_)
+    return addError("Token too long to be unsigned integer.", token, current);
+  for (; current < token.end_; ++current) {
+    Char c = *current;
+    if (c >= 'a')
+      c -= 'a' - 10;
+    else if (c >= 'A')
+      c -= 'A' - 10;
+    else if (c >= '0')
+      c -= '0';
+    else
+      return addError("Contains non-hexadecimal digits.", token, current);
```

Comment:
seems that this wouldn't catch nonhex characters like 'z'.

Reply:
Hmm... I think I had assumed that token.start/token.end were being set correctly by `readHexadecimal()`. It would make the method more robust and flexible to check both bounds though. Will fix.